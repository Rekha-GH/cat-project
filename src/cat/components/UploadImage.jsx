import React from "react";
import { useHistory } from "react-router-dom";
import ErrorMessage from "../../common/components/ErrorMessage";
import { upload } from "../model/ImageModel";

const DEFAULT_CLASSNAME = "upload-image";
const UploadImage = React.memo((props) => {
  const history = useHistory();
  const [loading, setLoading] = React.useState(false);
  const onChange = React.useCallback(
    (e) => {
      setLoading(true);

      const files = Array.from(e.target.files);
      const formData = new FormData();
      formData.append("sub_id", "");
      files.forEach((file, i) => {
        formData.append("file", file);
      });

      upload(formData)
        .then(() => {
          history.push("/");
        })
        .catch((e) => {
          setLoading(false);
        });
    },
    [setLoading, history]
  );

  return (
    <div className={DEFAULT_CLASSNAME}>
      <input disabled={loading} type="file" size="1" onChange={onChange} />
      {loading && <div>Uploding...</div>}
      <ErrorMessage />
    </div>
  );
});

export default UploadImage;
