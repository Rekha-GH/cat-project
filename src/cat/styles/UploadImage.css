.upload-image {
  padding: 15px;
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.upload-image > .upload-image__error {
  padding: 5px;
  border: 1px solid #ef5350;
  background: #e57373;
  border-radius: 5px;
}
