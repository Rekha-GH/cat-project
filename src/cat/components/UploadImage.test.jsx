import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import * as ReactRedux from "react-redux";
import UploadImage from "./UploadImage";

Enzyme.configure({ adapter: new Adapter() });

describe("UploadImage Test", () => {
  const useSelectorMock = jest.spyOn(ReactRedux, "useSelector");
  jest.spyOn(ReactRedux, "useDispatch");

  test("UploadImage Call Correctly", () => {
    useSelectorMock.mockReturnValue([]);
    const wrapper = shallow(<UploadImage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
