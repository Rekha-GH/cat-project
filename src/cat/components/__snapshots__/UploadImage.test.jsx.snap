// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImage Test UploadImage Call Correctly 1`] = `
<div
  className="upload-image"
>
  <input
    disabled={false}
    onChange={[Function]}
    size="1"
    type="file"
  />
  <Memo() />
</div>
`;
